lib/internal/main/watch_mode.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+const {
+  ArrayPrototypeConcat,
+  ArrayPrototypeMap,
+  ArrayPrototypeReduce,
+  ArrayPrototypeSlice,
+  ArrayPrototypeSome,
+  StringPrototypeSplit,
+  StringPrototypeStartsWith,
+} = primordials;
+const {
+  prepareMainThreadExecution,
+  markBootstrapComplete
+} = require('internal/process/pre_execution');
+const { getOptionValue } = require('internal/options');
+
+const { spawn } = require('child_process');
+const { watch } = require('fs/promises');
+const { setTimeout, clearTimeout } = require('timers');
+const { dirname, sep, resolve } = require('path');
+
+
+prepareMainThreadExecution(false);
+markBootstrapComplete();
+
+const kWatchedFiles = ArrayPrototypeMap(getOptionValue('--watch-file'), (file) => resolve(file));
+const args = ArrayPrototypeReduce(ArrayPrototypeConcat(
+  ArrayPrototypeSlice(process.execArgv),
+  ArrayPrototypeSlice(process.argv, 1),
+), (acc, flag, i, arr) => {
```

Comment:
https://github.com/nodejs/node/blob/81a68b37cfc9be0170e31349281b15b14f19959f/doc/contributing/primordials.md#L272-L288

I would love to add `ArrayPrototypeConcat` to the list of forbidden primordials in `node-core/avoid-prototype-pollution` lint rule, but would need someone to review/approve https://github.com/nodejs/node/pull/43476 first.

Reply:
Il give it a look